### ISSUE TITLE
Improve validation error messages 

### ISSUE THREAD

**Issue by fealho** [sdv-dev/SDV#936]
The `validate` method for `SingleTableMetadata` currently aggregates all error messages into one. This makes it so the traceback doesn't show where or why the error message occurred, so the only method of debugging the error message itself. Since some error messages are not specific enough, it can be hard to figure out what raised the error message in the first place. Below is an example of a few error messages the code currently raises:

```python3
'The following errors were found in the metadata:'
"\n\nMissing required values {'high_column_name'} in an Inequality constraint."
"\nInvalid values {'wrong_arg'} are present in an Inequality constraint."
"\n'value' must be an int or float."
"\n'primary_key' must be a string or tuple of strings."
"\nUnknown sequence key values {'col3'}. Keys should be columns that exist in the table."
"\n'alternate_keys' must be a list of strings or a list of tuples of strings."
"\nUnknown sequence index value {'col3'}. Keys should be columns that exist in the table."
"\n'sequence_index' and 'sequence_key' have the same value {'col3'}."
' These columns must be different.'
"\nInvalid values '(invalid1)' for categorical column 'col4'."
"\nInvalid order value provided for categorical column 'col5'."
" The 'order' must be a list with 1 or more elements."
"\nUnknown ordering method '' provided for categorical column 'col6'."
" Ordering method must be 'numerical_value' or 'alphabetical'."
"\nCategorical column 'col7' has both an 'order' and 'order_by' attribute."
' Only 1 is allowed.'
"\nInvalid value for 'representation' 'value' for column 'col8'."
"\nInvalid datetime format string '%1-%Y-%m-%d-%' for datetime column 'col9'."
"\nInvalid regex format string '[A-{6}' for text column 'col10'."
```
A few errors shown above are hard to decipher, e.g. `'value' must be an int or float.` doesn't really tell us where `'value'` comes from.

Basically, we should go through the errors and make them more specific.

**Comment by srinify** [comment]
Closing this issue as this is no longer the user experience of errors. SDV has come a long way since 2022! 👍
